analyses/snv-callers/scripts/01-calculate_vaf_tmb.R

```diff
@@ -0,0 +1,294 @@
+# Caller specific results directory path
+caller_results_dir <- file.path(root_dir, opt$output)
+
+# Make caller specific results folder
+if (!dir.exists(caller_results_dir)) {
+  dir.create(caller_results_dir, recursive = TRUE)
+}
+
+####################### File paths for files we will create ####################
+vaf_file <- file.path(caller_results_dir, paste0(opt$label, "_vaf.tsv"))
+region_annot_file <- file.path(caller_results_dir, paste0(opt$label, "_region.tsv"))
+tmb_file <- file.path(caller_results_dir, paste0(opt$label, "_tmb.tsv"))
+
+# Declare metadata file name for this caller
+metadata_file <- file.path(
+  caller_results_dir,
+  paste0(opt$label, "_metadata_filtered.tsv")
+)
+
+##################### Check for files if overwrite is FALSE ####################
+# If overwrite is set to FALSE, check if these exist before continuing
+if (!opt$overwrite) {
+  # Make a list of the output files
+  output_files <- c(vaf_file, region_annot_file, tmb_file)
+
+  # Find out which of these exist
+  existing_files <- file.exists(output_files)
+
+  # If all files exist; stop
+  if (all(existing_files)) {
+    stop(cat(
+      "Stopping; --overwrite is not being used and all output files already exist: \n",
+      vaf_file, "\n",
+      region_annot_file, "\n",
+      tmb_file
+    ))
+  }
+  # If some files exist, print a warning:
+  if (any(existing_files)) {
+    warning(cat(
+      "Some output files already exist and will not be overwritten unless you use --overwrite: \n",
+      paste0(output_files[which(existing_files)], "\n")
+    ))
+  }
+}
+
+########################### Set up this caller's data ##########################
+# Print progress message
+message(paste("Reading in", opt$maf, "MAF data..."))
+
+# Read in this MAF, skip the version number
+maf_df <- data.table::fread(opt$maf, skip = 1, data.table = FALSE)
+
+# Print progress message
+message(paste("Setting up", opt$label, "metadata..."))
+
+# Isolate metadata to only the samples that are in the datasets
+metadata <- readr::read_tsv(opt$metadata) %>%
+  dplyr::filter(Kids_First_Biospecimen_ID %in% maf_df$Tumor_Sample_Barcode) %>%
+  dplyr::distinct(Kids_First_Biospecimen_ID, .keep_all = TRUE) %>%
+  dplyr::arrange() %>%
+  dplyr::rename(Tumor_Sample_Barcode = Kids_First_Biospecimen_ID) %>%
+  readr::write_tsv(metadata_file)
+
+# Print out completion message
+message(paste("Filtered metadata file saved to: \n", metadata_file))
+
+# Make sure that we have metadata for all these samples.
+if (!all(unique(maf_df$Tumor_Sample_Barcode) %in% metadata$Tumor_Sample_Barcode)) {
+  stop("There are samples in this MAF file that are not in the metadata.")
+}
+
+################## Calculate VAF and set up other variables ####################
+# If the file doesn't exist or the overwrite option is being used, run this.
+if (any(c(!file.exists(vaf_file), opt$overwrite))) {
+
+  # Print out warning if this file is going to be overwritten
+  if (all(c(opt$overwrite, file.exists(vaf_file)))) {
```

Comment:
If the file exists, you must be overwriting, or you wouldn't be here:
```suggestion
  if (file.exists(vaf_file){
```

Reply:
Well, I have it written so if some files exist but others don't and overwrite is off it will only write the files that don't exist. Does that make sense? And also is there a better way to reflect that here?

Reply:
AH. I see what you are saying now. Yes. This is redundant. Will delete. Took a second for that to sink in.

Reply:
Yes, but If you play with the logic a bit, you will see that my version will still work.
There are 4 possible cases for each file

1.  file doesn't exist, overwrite off
2. file doesn't exist, overwrite on
3. file exists, overwrite off
4. file exists, overwrite on

You get to this line in cases 1, 2, and 4. And you only have to warn with case 4, so no need to check whether the file exists again.

Reply:
^ Cool. This is a good way to lay out the options.